tests/lib/workflows/test_search_by_expression.py

```diff
@@ -0,0 +1,104 @@
+from unittest.mock import patch, NonCallableMock, MagicMock
+import pytest
+
+from enums.query.sort_order import SortOrder
+from workflows.search_by_expression import search_by_expression
+
+
+@patch("workflows.search_by_expression.openstack_query")
+@patch("workflows.search_by_expression.QueryPresetsInteger")
+@pytest.mark.parametrize(
+    "output_type", ["to_html", "to_string", "to_objects", "to_props"]
```

Comment:
Just for understanding. This code allows testing of each of the types listed separately if I understand correctly?

Reply:
parametrize splits the test into 4 parts. Each test uses a different `output_type` parameter - set here `to_html`, `to_string` etc

Reply:
Thank you

Reply:
It's to save from writing each test separately - https://docs.pytest.org/en/7.1.x/example/parametrize.html